sources/Component/Collision/CollisionComponent.hpp

```diff
@@ -0,0 +1,67 @@
+//
+// Created by Louis Auzuret on 2021-05-20.
+//
+
+#pragma once
+
+#include "Models/Callback.hpp"
+#include "Models/Vector3.hpp"
+#include "Component/Component.hpp"
+#include "Entity/Entity.hpp"
+
+namespace BBM 
+{
+	class CollisionComponent : public WAL::Component
+	{
+		private:
+			//! @brief Bound size on all axis
+			Vector3f _bound;
```

Comment:
I think it will be easier to manipulate your bounds if this is a public field. A Vector3 has many helper functions and operators.